specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-04-01-preview/apimdeployment.json

```diff
@@ -475,6 +475,55 @@
         "x-ms-long-running-operation": true
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/migrateToStv2": {
+      "put": {
+        "tags": [
+          "ApiManagementService"
+        ],
+        "operationId": "ApiManagementService_MigrateToStv2",
+        "description": "Upgrades an API Management service to the Stv2 platform. This change is not reversible. This is long running operation and could take several minutes to complete.",
+        "x-ms-examples": {
+          "ApiManagementMigrateService": {
+            "$ref": "./examples/ApiManagementServiceMigrateToStv2.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The service was successfully migrated.",
```

Comment:
>migrated [](http://example.com/codeflow?start=57&length=8)

If the client loses the location header is there any way for it to know that this operation is ongoing ?

Reply:
What is the semantics in ARM if client loses the location header for any long running operation? 
Currently for any long running operation, we lock the resource and resource returns a transient state. The migration is going to be in place.

Reply:
If the resource is locked and the provisioningState is being modified that can be used as a fallback. But otherwise if the long running POST is happening independently without modifying the resource state then there is no other fallback.

Reply:
Yes, we lock the resource and provisioningState is modified to `Updating`